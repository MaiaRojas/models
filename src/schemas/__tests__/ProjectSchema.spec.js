const mongoose = require('mongoose/browser');
const { ProjectSchema } = require('../')(mongoose);

describe('ProjectSchema', () => {
  it('should fail validation when missing fields are provided', () => {
    const doc = new mongoose.Document({}, ProjectSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should fail validation when version not semver compatible', () => {
    const doc = new mongoose.Document({
      version: 'foo',
    }, ProjectSchema);
    return doc.validate()
      .catch(err => expect(err.errors.version.message).toBe('Invalid semver version foo'));
  });

  it('should successfully validate with proper values', (done) => {
    const doc = new mongoose.Document({
      slug: 'cipher',
      repo: 'Laboratoria/curricula-js',
      path: 'projects/01-cipher',
      version: '2.0.0',
      parserVersion: '1.1.1',
      createdAt: new Date(),
      prefix: 1,
      title: 'Cifrado CÃ©sar',
      rubric: '2',
      locale: 'es-ES',
      track: 'js',
      skills: {},
    }, ProjectSchema);
    doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
