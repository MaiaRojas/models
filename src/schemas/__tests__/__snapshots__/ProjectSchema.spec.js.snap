// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectSchema should fail validation when missing fields are provided 1`] = `
Object {
  "locale": [ValidatorError: Path \`locale\` is required.],
  "parserVersion": [ValidatorError: Path \`parserVersion\` is required.],
  "path": [ValidatorError: Path \`path\` is required.],
  "prefix": [ValidatorError: Path \`prefix\` is required.],
  "repo": [ValidatorError: Path \`repo\` is required.],
  "slug": [ValidatorError: Path \`slug\` is required.],
  "title": [ValidatorError: Path \`title\` is required.],
  "track": [ValidatorError: Path \`track\` is required.],
  "version": [ValidatorError: Path \`version\` is required.],
}
`;
